cookbook/pattern_matching.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Pattern Matching
+---
+
+# Pattern Matching
+
+## Using the `match` keyword
+
+Like many other languages, nu offers a [`match`](https://www.nushell.sh/commands/docs/match.html#frontmatter-title-for-core) keyword. Usually this is used as a slightly more ergonomic version of `if-else` statements if you have many branches
+
+```shell
+> [black red yellow green purple blue indigo] | each {|c|
+ match $c {
+  "black" => "classy"
+  "red" | "green" | "blue" => "fundamental"
+  "yellow" | "purple" => "vibrant"
+  _ => "innovative"
+ }
+}
+───┬────────────
+ 0 │ classy
+ 1 │ funamental
+ 2 │ vibrant
+ 3 │ funamental
+ 4 │ vibrant
+ 5 │ funamental
+ 6 │ innovative
+───┴────────────
+```
+
+The equivalent in `if-else` statements would be:
+
+```shell
+> [black red yellow green purple blue] | each {|c|
+ if ($c == "black") {
+  "classy"
+  } else if ( $c in ["red", "green", "blue"]) {
+   "fundamental"
+  } else if ( $c in ['yellow', "purple"]){
+   "vibrant"
+  }  else {
+   "innovative"
+  }
+ }
+}
+```
+
+As you can see you can also use comound expressions in match statements (in this case used with `|`). Also notice the `_` case at the end, this is called the default arm and is used.
+
+## Pattern matching on types
+
+You can use the [`describe`](https://www.nushell.sh/commands/docs/describe.html) command to get more info about the types of values. For example:
+
+```sh
+{one: 1 two: 2} | describe
+record<one: int, two: int>
+
+[{a: 1 b: 2} {a: 2 b:3 }] | describe
+table<a: int, b: int>
+```
+
+Together with `match` and some clever regex use you can do quite powerful type matching. For example, let's say we wanted to implement a `str append` function that would work on both strings and lists. On strings it would work as expected, on lists of strings, it should append the same string to each list. Using `match` one might do that like so:
```

Comment:
maybe? :innocent: 
```suggestion
Together with `match` and some clever regex use you can do quite powerful type matching. For example, let's say we wanted to implement a `str append` function that would work on both strings and lists. On strings it would work as expected, on lists of strings, it should append the same string to each element of the list. Using `match` one might do that like so:
```

Reply:
done in https://github.com/nushell/nushell.github.io/pull/1083/commits/0bad6309b64e456bc89f194c660d1d8899961839